From b02ae557c76a1a6ced6d2ca6a54c8f6978bcd22f Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Wed, 15 Apr 2015 12:13:57 +0800
Subject: [PATCH 355/360] atomisp2: Update global parameters at stream on

This patch updates the pending global parameters
at stream on where after css streams have been
created but before css is started.

Change-Id: I67d2966067a8bd82038acab786d012437eb20813
Fixes-Issue: #JIMINAN-28313
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/51308
Validation-Android: Wei Tao <wei.tao@intel.com>
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c    | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
index 35b1a74..c2427a1 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
@@ -1797,17 +1797,20 @@ static int atomisp_streamon(struct file *file, void *fh,
 	}
 
 	if (asd->params.css_update_params_needed) {
+		atomisp_apply_css_parameters(asd, &asd->params.css_param);
+		if (asd->params.css_param.update_flag.dz_config)
+			atomisp_css_set_dz_config(asd,
+				&asd->params.css_param.dz_config);
 		atomisp_css_update_isp_params(asd);
 		asd->params.css_update_params_needed = false;
+		memset(&asd->params.css_param.update_flag, 0,
+		       sizeof(struct atomisp_parameters));
 	}
 
 	ret = atomisp_css_start(asd, css_pipe_id, false);
 	if (ret)
 		goto out;
 
-
-	/* Make sure that update_isp_params is called at least once.*/
-	asd->params.css_update_params_needed = true;
 	asd->streaming = ATOMISP_DEVICE_STREAMING_ENABLED;
 	atomic_set(&asd->sof_count, -1);
 	atomic_set(&asd->sequence, -1);
-- 
1.9.1

